partiql-lang/src/main/kotlin/org/partiql/lang/eval/GlobalsCheck.kt

```diff
@@ -0,0 +1,21 @@
+package org.partiql.lang.eval
+
+interface GlobalsCheck {
```

Comment:
From what I can see, `GlobalsCheck` does not need to be made public. Its only use outside of this library is within the CLI, and all we do is initialize a `GlobalsCheck` using an `EvaluationSession` and pass it to `CompilerPipeline.Builder`.

So, an alternative to this, is to make this class `internal` and have the `CompilerPipeline.Builder` also build the `internal GlobalsCheck` using the passed `EvaluationSession`.

Even going beyond visibility -- it seems like all we need access to is an `EvaluationSession's` globals. Why not just use the session's globals directly? The `CompilerPipeline` also exposes `globalTypeBindings` as well.

If we remove public visibility, half the files touched don't actually need to be in this PR. If we remove it entirely, we may only need to make slight modifications.